util-secret/src/main/java/com/hpe/caf/secret/SecretUtil.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2015-2024 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.secret;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class SecretUtil
+{
+    private static final Logger LOG = LoggerFactory.getLogger(SecretUtil.class);
+    private static final String FILE_POSTFIX = "_FILE";
+    private static final String CAF_GET_SECRETS_FROM_ENV = "CAF_GET_SECRETS_FROM_ENV";
+    private static final String CAF_GET_SECRETS_FROM_FILE = "CAF_GET_SECRETS_FROM_FILE";
+
+    private SecretUtil()
+    {
+    }
+
+    /**
+     * Retrieves a secret value from various sources based on configuration:
+     * <ol>
+     *   <li>Environment variables (if CAF_GET_SECRETS_FROM_ENV is true or not set)</li>
+     *   <li>File content (if CAF_GET_SECRETS_FROM_FILE is true and path specified by environment variable with "_FILE" suffix)</li>
```

Comment:
@dermot-hardy Not sure if this will be an issue in practice but I noticed the following when testing, if I set the following (i.e. to switch to using the old file-based secrets approach):

USE_FILE_BASED_SECRETS=true
CAF_GET_SECRETS_FROM_FILE=false

I get this error, because export-file-based-secrets.sh tries to read the CAF_GET_SECRETS_FROM_FILE env as a file:

Oct 18 09:05:29 larry-k8spri02 k8s_archivecleanup-worker_archivecleanup-worker-64dd8f9488-dvggq_private_4f90059d-d60e-4866-93ae-a036af7d024a_3:  [2024-10-18 09:05:29.958Z #00E.??? INFO  -            -   ] export-file-based-secrets.sh: Reading CAF_GET_SECRETS_FROM_FILE (false)...
Oct 18 09:05:29 larry-k8spri02 k8s_archivecleanup-worker_archivecleanup-worker-64dd8f9488-dvggq_private_4f90059d-d60e-4866-93ae-a036af7d024a_3:  [2024-10-18 09:05:29.969Z #014.??? ERROR -            -   ] export-file-based-secrets.sh: Failed to export env variable CAF_GET_SECRETS_FROM, file false does not exist

This might be a problem if someone wants to switch from the new approach to the old approach? It works if you completely remove the CAF_GET_SECRETS_FROM_FILE env var (and let it default to false), but that would not be obvious to a user.

Maybe the env vars should be renamed to?

CAF_ENV_SECRETS_ENABLED
CAF_FILE_SECRETS_ENABLED

Reply:
I see.  I guess we should rename them now when we're still early enough in the process.  Should we put the `_ENABLE` earlier?  I.e. `CAF_ENABLE_ENV_SECRETS` and `CAF_ENABLE_FILE_SECRETS`?